apps/plantd-proxy/routes/utils.go

```diff
@@ -15,38 +19,142 @@ type ErrorResponse struct {
 // The handler function retrieves the sample dataset based on the provided namespace and dataset name.
-// It calls the proxy.GetSampleDataset function and writes the dataset to the response.
-// If there is an error during the retrieval process, it returns an HTTP 500 status with an error message.
-func GetSampleDataset(client client.Client) http.HandlerFunc {
+// It calls the proxy.GetSampleDataSet function and writes the dataset to the response.
+// If there is an error during the retrieval process, it responds an HTTP 500 status with an error message.
+func getSampleDataSet(client client.Client) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
 		namespace := chi.URLParam(r, "namespace")
-		datasetName := chi.URLParam(r, "name")
-		if fileFormat, bytes, err := proxy.GetSampleDataset(ctx, client, namespace, datasetName); err != nil {
+		name := chi.URLParam(r, "name")
+		if fileExt, bytes, err := proxy.GetSampleDataSet(ctx, client, namespace, name); err != nil {
+			w.Header().Set("Content-Type", "application/json")
 			w.WriteHeader(http.StatusInternalServerError)
-			json.NewEncoder(w).Encode(ErrorResponse{Message: "while getting sample DataSet: " + err.Error()})
+			json.NewEncoder(w).Encode(ErrorResponse{Message: "while getting sample dataset: " + err.Error()})
 			return
 		} else {
-			contentDisposition := "attachment; filename=example." + fileFormat
+			contentDisposition := fmt.Sprintf("attachment; filename=sample_%s_%s.%s", namespace, name, fileExt)
 			w.Header().Set("Content-Disposition", contentDisposition)
-			contentType := "application/" + fileFormat
-			w.Header().Set("Content-Type", contentType)
+			w.Header().Set("Content-Type", "application/octet-stream")
 			w.WriteHeader(http.StatusOK)
+			bytes.WriteTo(w)
+		}
+	}
+}
 
-			if _, err := bytes.WriteTo(w); err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				json.NewEncoder(w).Encode(ErrorResponse{Message: "while writing response body: " + err.Error()})
-				return
-			}
+type CheckHTTPHealthRequest struct {
+	URL string `json:"url,omitempty"`
+}
+
+// checkHTTPHealth returns an HTTP handler function for checking health status of a URL using HTTP protocol.
+// The handler function retrieves the sample dataset based on the provided namespace and dataset name.
+// It calls utils.CheckHTTPHealth to make a request to the designated URL. Upon receiving an HTTP non-200 response,
+// it responds an HTTP 500 status with an ErrorResponse in JSON.
+func checkHTTPHealth() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusBadRequest)
+			json.NewEncoder(w).Encode(ErrorResponse{Message: "while reading request body: " + err.Error()})
+			return
+		}
+		data := CheckHTTPHealthRequest{}
+		err = json.Unmarshal(body, &data)
```

Comment:
There are lots of places where you passing the address of the slice as the parameter. Could you elaborate the reason?

Reply:
Fixed. The initial purpose of passing a pointer was to avoid copying and improve performance. However, since the essence of a slice is a pointer, it is redundant to pass a pointer of a slice.